makefile

```diff
@@ -517,8 +517,9 @@ $(ALL_CLEAN_SUBDIRS):
 	$(MAKE) -C $(BLD)$(@D) $(@F)
 
 clean: $(addsuffix /clean,$(EM_DIRS) $(TEST_DIRS))
-	rm -rf $(addprefix $(BLD),$(foreach d,$(DIRS),$(d)/*{.o,~}) \
-	   *~ libmfem.* deps.mk)
+	rm -f $(addprefix $(BLD),$(foreach d,$(DIRS),$(d)/*.o))
+	rm -f $(addprefix $(BLD),$(foreach d,$(DIRS),$(d)/*~))
+	rm -rf $(addprefix $(BLD),*~ libmfem.* deps.mk)
```

Comment:
Which one is expected to be a directory here?

Reply:
Shared debug build on mac with CUDA enabled creates directory called `libmfem.4.0.1.dylib.dSYM`.